src/field_5x52_impl.h

```diff
@@ -667,4 +667,32 @@ static void secp256k1_fe_inv_var(secp256k1_fe *r, const secp256k1_fe *x) {
 #endif
 }
 
+static int secp256k1_fe_jacobi_var(const secp256k1_fe *x) {
+    secp256k1_fe tmp;
+    secp256k1_modinv64_signed62 s;
+    int ret;
+
+    tmp = *x;
+    secp256k1_fe_normalize_var(&tmp);
+    secp256k1_fe_to_signed62(&s, &tmp);
+    ret = secp256k1_jacobi64_maybe_var(&s, &secp256k1_const_modinfo_fe);
+    if (ret == -2) {
+        /* secp256k1_jacobi64_maybe_var failed to compute the Jacobi symbol. Fall back
+         * to computing a square root. This should be extremely rare with random
+         * input. */
+        secp256k1_fe dummy;
+        ret = 2*secp256k1_fe_sqrt(&dummy, &tmp) - 1;
```

Comment:
Should instead have a function  `fe_is_square_var`? 

I think this will be better because it
 * avoids the return value translations here and below 
 * is more ergonomic to use (I think, remember that we had it this in the old version of BIP340 too),
 * and it aligns with the uses in ellswift: in the corresponding PR, we only test `>=0` and `<0`, so we actually don't even need the special casing for `==0` currently.

This is related to my comment below about the `jacp` function argument.

Reply:
makes sense